### PR TITLE
Implement textDocument/prepareRename

### PR DESCRIPTION
**Pull request by cannorin**
TBD:
- [x] Does it need a test?

### CODE REVIEW

File: ocaml-lsp-server/src/ocaml_lsp_server.ml [hunk 1]
```diff
@@ -35,7 +35,10 @@ let initializeInfo : InitializeResult.t =
       ~referencesProvider:(`Bool true) ~documentHighlightProvider:(`Bool true)
       ~documentFormattingProvider:(`Bool true)
       ~selectionRangeProvider:(`Bool true) ~documentSymbolProvider:(`Bool true)
-      ~renameProvider:(`Bool true) ()
+      ~renameProvider:
+        (`RenameOptions
+          { prepareProvider = Some true; workDoneProgress = None })
```
Comment (rgrinberg) [inline ocaml-lsp-server/src/ocaml_lsp_server.ml]: Please use the constructor for RenameOptions. This way, our code will not be broken once new fields are added

Comment (cannorin) [reply ocaml-lsp-server/src/ocaml_lsp_server.ml]: For unknown reasons `module RenameOptions` is not included in the `Import` module and it is failing to compile:
Screenshot from 2020-03-18 17-35-28

Comment (cannorin) [reply ocaml-lsp-server/src/ocaml_lsp_server.ml]: It's not just in the `ocaml-lsp-server/src/import.ml`. It is inconvenient for me to maintain it by hand btw.

**Comment by rgrinberg** [comment]
Merged manually. Thanks.

**Pull request closed without merging**